pwiz_tools/Skyline/Executables/SharedBatch/LaunchBatch/Program.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Deployment.Application;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Windows.Forms;
+using LaunchBatch.Properties;
+
+namespace LaunchBatch
+{
+    static class Program
+    {
+        /// <summary>
+        /// The main entry point for the application.
+        /// </summary>
+        [STAThread]
+        static void Main(string[] args)
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+
+            var appReferencePath = args[0];
+
+            // join remaining args (will be more than one if filepath has spaces)

```

Comment:
This doesn't sound right. Normally paths with spaces appear as a single argument value.
If they are coming in as more than one argument, then it usually means that someone else was supposed to put quotes around it.
Trying to reconstruct the original value by adding spaces in between the arguments that you got is not going to work if the path happened to have two spaces in a row in it.

Reply:
That makes sense, I didn't have enough escaped quotes when I added it to the registry. Fixed it now and the path comes in as one argument, thanks!